src/mlpack/methods/ne/gene.hpp

```diff
@@ -0,0 +1,186 @@
+/**
+ * @file gene.hpp
+ * @author Bang Liu
+ *
+ * Definition of the LinkGene and NeuronGene class.
+ */
+#ifndef MLPACK_METHODS_NE_GENE_HPP
+#define MLPACK_METHODS_NE_GENE_HPP
+
+#include <mlpack/core.hpp>
+
+namespace mlpack {
+namespace ne {
+
+/**
+ * This is enumeration of neuron types.
+ */
+enum NeuronType {
+  NONE = 0,
+  INPUT,
+  BIAS,
+  HIDDEN,
+  OUTPUT
+};
+
+/**
+ * This is enumeration of activation function types.
+ */
+enum ActivationFuncType {
+    SIGMOID = 0,
+    TANH,
+    LINEAR,
+    RELU,
+};
+
+/**
+ * This class defines a link gene.
+ */
+class LinkGene {
+ public:
+  // Default constructor.
+  LinkGene() {}
+
+  // Parametric constructor.
+  LinkGene(unsigned int fromNeuronId,
+  	       unsigned int toNeuronId,
+  	       unsigned int innovationId,
+  	       double weight):
+    aFromNeuronId(fromNeuronId),
+    aToNeuronId(toNeuronId),
+    aInnovationId(innovationId),
+    aWeight(weight)
+  {}
+
+  // Copy constructor.
+  LinkGene(const LinkGene& linkGene) {
+  	aFromNeuronId = linkGene.aFromNeuronId;
+  	aToNeuronId = linkGene.aToNeuronId;
+  	aInnovationId = linkGene.aInnovationId;
+  	aWeight = linkGene.aWeight;
+  }
+
+  // Destructor.
+  ~LinkGene() {}
+
+  // Get aFromNeuronId.
+  unsigned int FromNeuronId() const { return aFromNeuronId; }
+
+  // Get aToNeuronId.
+  unsigned int ToNeuronId() const { return aToNeuronId; }
+
+  // Get aInnovationId.
+  unsigned int InnovationId() const { return aInnovationId; }
+
+  // Set aWeight.
+  void Weight(double weight) { aWeight = weight; }
+
+  // Get aWeight.
+  double Weight() const { return aWeight; }
+
```

Comment:
OK,  got it! I will update my code style later in according with it!

On Thursday, 9 June 2016, Ryan Curtin <notifications@github.com> wrote:

> In src/mlpack/methods/ne/gene.hpp
> <https://github.com/mlpack/mlpack/pull/686#discussion_r66444075>:
>
> > +
> > +  // Get aFromNeuronId.
> > +  unsigned int FromNeuronId() const { return aFromNeuronId; }
> > +
> > +  // Get aToNeuronId.
> > +  unsigned int ToNeuronId() const { return aToNeuronId; }
> > +
> > +  // Get aInnovationId.
> > +  unsigned int InnovationId() const { return aInnovationId; }
> > +
> > +  // Set aWeight.
> > +  void Weight(double weight) { aWeight = weight; }
> > +
> > +  // Get aWeight.
> > +  double Weight() const { return aWeight; }
> > +
>
> I think that the naming convention section of the mlpack style guide was a
> little unclear, it suggested that parameters should have an a in front of
> them but that's not the case, just the examples used there were bad. So I
> would go with weight instead of aWeight. I updated the naming convention
> section, hopefully it is more clear now; let me know if anything still
> needs to be revised. :)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mlpack/mlpack/pull/686/files/3c8aa62b951f029b3883e9baef1ea556ef5af2d3#r66444075>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AD9tIdBQdgeN2HnGki74ONIL-g9orOKkks5qKBqjgaJpZM4IwJa6>
> .
>